azext_edge/edge/providers/rpsaas/adr/assets.py

```diff
@@ -116,386 +116,707 @@ def create(
             product_code=product_code,
             serial_number=serial_number,
             software_revision=software_revision,
-            dp_publishing_interval=dp_publishing_interval,
-            dp_sampling_interval=dp_sampling_interval,
-            dp_queue_size=dp_queue_size,
+            ds_publishing_interval=ds_publishing_interval,
+            ds_sampling_interval=ds_sampling_interval,
+            ds_queue_size=ds_queue_size,
             ev_publishing_interval=ev_publishing_interval,
             ev_sampling_interval=ev_sampling_interval,
             ev_queue_size=ev_queue_size,
         )
+        # discovered
+        if discovered:
+            self.ops = self.discovered_ops
+            properties.pop("enabled", None)
+            properties["version"] = 1
+            properties["discoveryId"] = "discoveryid1"
 
         asset_body = {
             "extendedLocation": extended_location,
-            "location": location,
+            "location": location or cluster_location,
             "properties": properties,
             "tags": tags,
         }
-        poller = self.resource_client.resources.begin_create_or_update(
+        return self.ops.begin_create_or_replace(
+            resource_group_name,
+            asset_name,
+            resource=asset_body
+        )
+
+    def delete(self, asset_name: str, resource_group_name: str):
+        self.show(
+            asset_name=asset_name,
             resource_group_name=resource_group_name,
-            resource_provider_namespace=self.provider_namespace,
-            parent_resource_path="",
-            resource_type=self.resource_type,
-            resource_name=asset_name,
-            api_version=self.api_version,
-            parameters=asset_body
+            check_cluster=True
         )
-        return poller
+        return self.update_ops.begin_delete(
+            resource_group_name,
+            asset_name,
+        )
+
+    def show(
+        self, asset_name: str, resource_group_name: str, check_cluster: bool = False
+    ) -> dict:
+        # TODO: re-add try except when discovered is exposed
+        # try:
+        asset = self.ops.get(
+            resource_group_name=resource_group_name, asset_name=asset_name
+        )
+        self.update_ops = self.ops
+        # except ResourceNotFoundError:
```

Comment:
With this code uncommented, every discovered asset is going to take +2 API calls?

Reply:
well since I want to have the discovered assets + assets in the same class and share methods (cause they are almost the same props wise)

and....

this saves the user from remembering which asset is a discovered asset vs normal (and then having to put in --discovered or something)

given asset name + rg
1, show - see if it is an asset first
2. if 1 fails - see if it is a discovered asset
3. do the update/fun stuff if needed

Reply:
And if they only want to operate on the discovered asset if both exist ?

Reply:
see other discovered comment